### ISSUE TITLE
Questions about the pretrained models and how they are trained

### ISSUE THREAD

**Issue by datistiquo** [UKPLab/sentence-transformers#725]
I have to say, that is quote astonishing how fast here everythings moves forward, and also the community!
If you pause for 1 month, you feel you get lost with the whole progress. :)

My experiments with finetuning base bert model are good, but they can be better. That is why I will give the multilingual models a shot as most write they get better results (for german).

I want to use such a pretrained model and finetune it on my german data. Where can I find the exact scripts where the original models like paraphrase-xlm-r-multilingual-v1 or distilbert-multilingual-nli-stsb-quora-ranking are trained with all belonging steps and loss functions used? I cannot find that. The same also for this model:
https://huggingface.co/T-Systems-onsite/cross-en-de-roberta-sentence-transformer

I think for fientuning the cross-en-de-roberta-sentence-transformer there are already some scripts?

Can I actually finetune such a model if I just have binary labels and no scores (as scores are a little bit unrealistic in praxis)

I think models like distilbert-multilingual-nli-stsb-quora-ranking are trained in a traditional bi-encoder fashion?
Is it possible and a good idea to take such a pretrained model which was trained in bi encoder fashion and use it in a cross encoder for fientuning (or vice versa)?

**Comment by nreimers** [comment]
For training see the examples here:
https://github.com/UKPLab/sentence-transformers/tree/master/examples/training

Multi-lingual cross-encoders are sadly not yet available.

> Note: The model don’t work for question similarity. The question How to learn Java and How to learn Python will get a low score, as these questions are not duplicates. For question similarity, the respective bi-encoder trained on the Quora dataset yields much more meaningful results.

> I thought cross-encoders would capture semantic simalirity better than bi-encoders?

Cross-Encoders are more powerful than bi-encoders, that is correct.

but the mentioned cross-encoder for duplicate questions will output a 1 if the the question pair is a duplicate and a score close to 0 if it is not a duplicate.
So it correctly outputs:
cross-encoder( How to learn Java, How to learn Python) -> score close to 0
cross-encoder( How to learn Java, How many people live in London) -> score close to 0

Both questions pairs are non-duplicates, so both get correctly a score close to 0 from the cross-encoder.

The bi-encoder instead computes the distance in vector space, hence, there the notation of similarity is better:
 bi-encoder( How to learn Java, How to learn Python)  > bi-encoder( How to learn Java, How many people live in London)
